src/java/org/apache/cassandra/replication/MutationJournal.java

```diff
@@ -73,7 +122,40 @@ public void shutdownBlocking()
 
     public RecordPointer write(ShortMutationId id, Mutation mutation)
     {
-        return journal.blockingWrite(id, mutation);
+        // TODO (required): why are we using blocking write here? We can/should wait for completion on `close` of WriteContext.
```

Comment:
I think we do the same with vanilla commit log. Here we need to mutation to be fully durable before we update/share witnessed state.

Reply:
Yes, we do the same in vanilla commit log. But my thinking was that we can do at least part of work while mutation is fsynced. I left a TODO to not to forget to discuss this with you.